#161255317 Social authentication

AnguleMathias opened the pull request:
#### What does this PR do?
* Add's social login to the application
* It also adds action types for a successful login.

#### Description of tasks to be completed?
* Users should be able to Login via social login to the application

#### How should this be manually tested?
* Run npm install to install all dependencies
* Run npm start
* Enter the url `http://localhost:3000/login` in your browser.
* click the social login buttons

#### What are the relevant pivotal tracker stories?
* #161255317 - [Users should be able to Login via Google Facebook or Twitter](https://www.pivotaltracker.com/story/show/161255317)

3N61N33R committed:
feature(registration): user signup

-user successfully registers
-send verfication email to use

[feature #161255313]

AnguleMathias committed:
Feature(Social Authentication): Users should be able to login through social authentication

- Google and facebook authentication

[Feature #161255317]

AnguleMathias committed:
Feature(Social Authentication): Users should be able to login through social authentication

- Google and facebook authentication

[Feature #161255317]

AnguleMathias committed:
Feature(Social Authentication): Users should be able login via social
- test user actions

[Feature #161255317]

AnguleMathias committed:
Feature(Social Authentication): Users should be able login via social
- test user actions

[Feature #161255317]

3N61N33R committed:
feature(registration): register user

-add tests
-update styling

[feature #161255313]

AnguleMathias committed:
Feature(Social Authentication): Users should be able to login through social authentication

- User actions test update

[Feature #161255317]

AnguleMathias committed:
Feature(Social Authentication): Users should be able to login through social authentication

- User actions test update

[Feature #161255317]

njeri-ngigi committed:
Merge pull request #11 from andela/ft-user-signup-161255313

#161255313 Users should be able Signup/Register to Authors Haven

AnguleMathias committed:
Feature(Social Authentication): Users should be able to login through social authentication

- Button(facebook,google) tests

[Feature #161255317]

AnguleMathias committed:
Feature(Social Authentication): Users should be able to login through social authentication

- Button(facebook,google) tests

[Feature #161255317]

meshnesh commented:
@AnguleMathias your, P.R is bringing in 6 issues of code duplication as stated in code climate, here is the link. https://codeclimate.com/github/andela/ah-codeofduty-frontend/pull/10, look into solving them before the P.R can be merged.

maxwellgithinji committed:
#161255316 Users should be able to Login to Authors Haven (#12)

* Feature(Login): Users should be able to Login to Authors Haven
- ensure GIVEN user successfully logs in.
WHEN user passes appropriate login parameters
THEN user should be able to login.
- ensure descriptive validation errors are present
- ensure a link for users to Reset their passwords is present
[Feature #161255316]

* Feature(Login): Users should be able to Login to Authors Haven
- ensure to rectify test files and anchor proprties
[Feature #161255316]

* Feature(Login): Users should be able to Login to Authors Haven
- ensure to rectify test files
[Feature #161255316]

* Feature(Login): Users should be able to Login to Authors Haven
- ensure to change file structure
[Feature #161255316]

* Feature(Login): Users should be able to Login to Authors Haven
- ensure to alert messages are properly displayed
[Feature #161255316]

* Feature(Login): Users should be able to Login to Authors Haven
- ensure GIVEN user successfully logs in.
WHEN user passes appropriate login parameters
THEN user should be able to login.
- ensure descriptive validation errors are present
- ensure a link for users to Reset their passwords is present
[Feature #161255316]

* Feature(Login): Users should be able to Login to Authors Haven
- ensure to rectify test files and anchor proprties
[Feature #161255316]

* Feature(Login): Users should be able to Login to Authors Haven
- ensure to rectify test files
[Feature #161255316]

* Feature(Login): Users should be able to Login to Authors Haven
- ensure to change file structure
[Feature #161255316]

* Feature(Login): Users should be able to Login to Authors Haven
- ensure to alert messages are properly displayed
[Feature #161255316]

* Feature(Login): Users should be able to Login to Authors Haven
- ensure to resolve merge conflicts and integrate code
[Feature #161255316]

* Feature(Login): Users should be able to Login to Authors Haven
- ensure to resolve sticky modal
[Feature #161255316]

* Feature(Login): Users should be able to Login to Authors Haven
- ensure to remove redundant files
[Feature #161255316]

AnguleMathias committed:
Feature(Social Authentication): Users should be able to login through social authentication

- social login component tests

[Feature #161255317]

AnguleMathias committed:
Feature(Social Authentication): Users should be able to login through social authentication

- social login component tests

[Feature #161255317]

AnguleMathias committed:
Feature(Social Authentication): Users should be able to login through social authentication

- Rebase commit

[Feature #161255317]
